srml/contracts/src/wasm/prepare.rs

```diff
@@ -227,14 +227,16 @@ impl<'a> ContractModule<'a> {
 			};
 
 			// Then check the signature.
-			// Both "call" and "deploy" has a () -> () function type.
+			// Both "call" and "deploy" has a [] -> [] or [] -> [i32] function type.
```

Comment:
Well, I am not sure that we need to do this. Although I don't see any problems with it right now, I think it might complicate things in future, while it doesn't provide much benefit (maybe I am missing something though).

Let's make it strictly `[] -> [i32]`

Reply:
Yeah, the only benefits are that 1) it's backwards compatible, which is maybe a good idea until ink! is updated, and 2) it makes the tests, which use hand-written Wasm a bit simpler.

My preference would be to keep in this PR and remove the `[] -> []` signature in a follow-up, but I could remove it in this one as well if you prefer.

Reply:
Ah, no. That works! Just add a comment.